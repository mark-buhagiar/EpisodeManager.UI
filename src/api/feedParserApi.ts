import axios from 'axios';
import ApiConfig from '../config/apiConfig';
import ParseHistoryListing from '../models/ParseHistoryListing';
import ParseHistoryDetails from '../models/ParseHistoryDetails';
import moment from 'moment';

const buildEndpoint = (endpoint: string): string => ApiConfig.feedParserApiBase + endpoint;

const feedParserMSEndpoints = {
    syncWithShowRss: buildEndpoint('sync/SyncWithShowRss'),
    forceSyncParseHistory: buildEndpoint('sync/ForceSyncParseHistory'),
    getParseHistoryBetweenDates: buildEndpoint('ParseHistory/GetParseHistoryBetweenDates'),
    getParseHistoryDetails: buildEndpoint('ParseHistory/GetParseHistoryDetails'),
};

export async function syncWithShowRss(): Promise<void> {
    try {
        await axios.post(feedParserMSEndpoints.syncWithShowRss);
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

function convertDates(listing: ParseHistoryListing): ParseHistoryListing {
    return {
        ...listing,
        retrievedOn: moment.utc(listing.retrievedOn).toDate(),
        lastSyncedOn: moment.utc(listing.lastSyncedOn).toDate(),
    };
}

export async function getParseHistoryBetweenDates(startDate: Date, endDate: Date): Promise<ParseHistoryListing[]> {
    try {
        const params = {
            startDate,
            endDate,
        };
        const parseHistory = await axios.get<ParseHistoryListing[]>(feedParserMSEndpoints.getParseHistoryBetweenDates, {
            params,
        });

        return parseHistory.data.map(convertDates);
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

export async function getParseHistoryDetails(id: number): Promise<ParseHistoryDetails> {
    try {
        const params = {
            parseHistoryId: id,
        };
        const parseHistoryDetails = await axios.get<ParseHistoryDetails>(feedParserMSEndpoints.getParseHistoryDetails, {
            params,
        });

        return parseHistoryDetails.data;
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}
export async function forceSyncParseHistory(id: number): Promise<void> {
    try {
        const params = {
            parseHistoryId: id,
        };

        await axios.post(feedParserMSEndpoints.forceSyncParseHistory, null, {
            params,
        });
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}
