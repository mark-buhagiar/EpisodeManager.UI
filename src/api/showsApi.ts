import axios from 'axios';
import Episode from '../models/Episode';
import ApiConfig from '../config/apiConfig';

const buildEndpoint = (endpoint: string): string => ApiConfig.showsApiBase + endpoint;

const showsMSEndpoints = {
    getForCurrentUserBetweenDates: buildEndpoint('episodes/getForCurrentUserBetweenDates'),
};

export async function getForCurrentUserBetweenDates(startDate: Date, endDate: Date): Promise<Episode[]> {
    try {
        const params = {
            startDate: startDate,
            endDate: endDate,
        };
        const result = await axios.get<Episode[]>(showsMSEndpoints.getForCurrentUserBetweenDates, { params });
        return result.data;
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}
