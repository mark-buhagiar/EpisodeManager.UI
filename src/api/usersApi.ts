import axios from 'axios';
import ApiConfig from '../config/apiConfig';
import EpisodePreferences from '../models/EpisodePreferences';
import Show from '../models/Show';

const buildEndpoint = (endpoint: string): string => ApiConfig.usersApiBase + endpoint;

const usersMSEndpoints = {
    addEpisodeDownloaded: buildEndpoint('subscriptions/AddEpisodeDownloaded'),
    getUserSubscriptionsForListing: buildEndpoint('subscriptions/GetUserSubscriptionsForListing'),
    unsubscribeFromShow: buildEndpoint('subscriptions/UnsubscribeFromShow'),
    subscribeToShow: buildEndpoint('subscriptions/SubscribeToShow'),
    setUserEpisodePreferences: buildEndpoint('users/setUserEpisodePreferences'),
    getUserEpisodePreferences: buildEndpoint('users/getUserEpisodePreferences'),
};

export async function addEpisodeDownloaded(episodeId: number): Promise<void> {
    try {
        const params = {
            episodeId,
        };
        await axios.put(usersMSEndpoints.addEpisodeDownloaded, null, { params });
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

export async function getUserSubscriptionsForListing(): Promise<Show[]> {
    try {
        const result = await axios.get(usersMSEndpoints.getUserSubscriptionsForListing);
        return result.data;
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

export async function unsubscribeFromShow(showId: number): Promise<void> {
    try {
        const params = {
            showId,
        };
        await axios.patch(usersMSEndpoints.unsubscribeFromShow, { params });
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

export async function subscribeToShow(showIds: number[]): Promise<void> {
    try {
        const params = {
            showIds,
        };
        await axios.put(usersMSEndpoints.subscribeToShow, null, { params });
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

export async function setUserEpisodePreferences(preferences: EpisodePreferences): Promise<void> {
    try {
        await axios.post(usersMSEndpoints.setUserEpisodePreferences, preferences);
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}

export async function getUserEpisodePreferences(): Promise<EpisodePreferences> {
    try {
        const response = await axios.get(usersMSEndpoints.getUserEpisodePreferences);
        return response.data;
    } catch (exception) {
        console.log(exception);
        throw exception;
    }
}
