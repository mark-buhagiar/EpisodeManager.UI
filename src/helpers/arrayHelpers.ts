type Dictionary<T> = {
    [key: string]: T[];
};

export function groupBy<T>(items: T[], key: (item: T) => string): Dictionary<T> {
    return items.reduce(
        (result, item) => ({
            ...result,
            [key(item)]: [...(result[key(item)] || []), item],
        }),
        {} as Dictionary<T>,
    );
}
