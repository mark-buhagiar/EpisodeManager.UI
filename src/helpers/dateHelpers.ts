import moment, { Moment } from 'moment';

const DATE_FORMAT = 'D MMMM YYYY';
const DATE_AND_TIME_FORMAT = 'D MMMM YYYY HH:mm';

export function toDateFormat(date: Moment): string {
    return date.format(DATE_FORMAT);
}

export function toDateAndTimeFormat(date: Moment): string {
    return date.format(DATE_AND_TIME_FORMAT);
}

export function formStringToDateFormat(date: string | undefined): string {
    if (date === undefined || date === null) return 'N/A';
    return toDateFormat(moment(date));
}

export function getDatesInRange(start: Moment, end: Moment): Date[] {
    const dateArray = [];
    let currentDate = moment(start);
    while (currentDate.isBefore(end)) {
        dateArray.push(currentDate.toDate());
        currentDate = currentDate.add(1, 'day');
    }
    return dateArray;
}
