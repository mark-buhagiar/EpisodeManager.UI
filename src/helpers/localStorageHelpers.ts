import LocalStorageItem from '../models/LocalStorageItem';

function findLocalStorageItems(query: string): LocalStorageItem[] {
    let i;
    const results = [];
    for (i in localStorage) {
        if (localStorage.hasOwnProperty(i)) {
            if (i.match(query) || (!query && typeof i === 'string')) {
                const value = localStorage.getItem(i);
                if (value !== null) {
                    const parsedValue = JSON.parse(value);
                    results.push({ key: i, val: parsedValue });
                }
            }
        }
    }
    return results;
}

export function removeItemWithRegexKey(key: string): void {
    try {
        localStorage.removeItem(findLocalStorageItems(key)[0].key);
    } catch {
        console.log('LocalStorage key not found');
    }
}
