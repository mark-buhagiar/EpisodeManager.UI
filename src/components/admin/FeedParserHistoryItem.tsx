import {
    Divider,
    ExpansionPanel,
    ExpansionPanelActions,
    ExpansionPanelDetails,
    ExpansionPanelSummary,
} from '@material-ui/core';
import React, { useState } from 'react';
import * as feedParserApi from '../../api/feedParserApi';
import ParseHistoryDetails from '../../models/ParseHistoryDetails';
import ParseHistoryListing from '../../models/ParseHistoryListing';
import { ButtonProps } from '../common/button/Button';
import ButtonGroup from '../common/button/ButtonGroup';
import ComponentLoading from '../loading/ComponentLoading';
import { toDateAndTimeFormat } from '../../helpers/dateHelpers';
import moment from 'moment';

interface Props {
    parseHistoryItem: ParseHistoryListing;
}

const FeedParserHistoryItem: React.FC<Props> = ({ parseHistoryItem: summary }: Props): JSX.Element => {
    const [details, setDetails] = useState<ParseHistoryDetails | null>(null);
    const [forceResyncEnabled, setForceResyncEnabled] = useState<boolean>(false);

    async function handleForceResyncClicked(): Promise<void> {
        if (details === null) return;
        setForceResyncEnabled(false);
        await feedParserApi.forceSyncParseHistory(details.id);
        setForceResyncEnabled(true);
    }

    const buttons = [
        { label: 'Force Re-sync', onClick: handleForceResyncClicked, enabled: forceResyncEnabled },
    ] as ButtonProps[];

    async function handlePanelExpansion(event: React.ChangeEvent<{}>, expanded: boolean): Promise<void> {
        if (!expanded) return;
        if (details !== null) return; // this is a re-expansion

        setDetails(await feedParserApi.getParseHistoryDetails(summary.id));
        setForceResyncEnabled(true);
    }

    return (
        <ExpansionPanel TransitionProps={{ unmountOnExit: true }} onChange={handlePanelExpansion}>
            <ExpansionPanelSummary expandIcon={<span className="material-icons">expand_more</span>}>
                <div className="parse-history-summary">
                    <div className="date" title="Last synced on">
                        {toDateAndTimeFormat(moment(summary.lastSyncedOn))}
                    </div>
                    <div className="episode-count" title="Episode count">
                        {summary.episodeCount}
                    </div>
                    <div className="status">{summary.success ? 'Success' : 'Failure'}</div>
                </div>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails>
                <div className="parse-history-details">
                    {details === null ? (
                        <ComponentLoading />
                    ) : (
                        <ul>
                            {details.episodeTitles
                                .sort((a, b) => (a > b ? 1 : -1))
                                .map((title) => (
                                    <li key={title}>{title}</li>
                                ))}
                        </ul>
                    )}
                </div>
            </ExpansionPanelDetails>
            <Divider />
            <ExpansionPanelActions>
                <ButtonGroup buttons={buttons} />
            </ExpansionPanelActions>
        </ExpansionPanel>
    );
};

export default FeedParserHistoryItem;
