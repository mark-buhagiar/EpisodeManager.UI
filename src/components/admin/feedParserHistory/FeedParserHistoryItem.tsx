import {
    Divider,
    ExpansionPanel,
    ExpansionPanelActions,
    ExpansionPanelDetails,
    ExpansionPanelSummary,
} from '@material-ui/core';
import moment from 'moment';
import React, { useState } from 'react';
import * as feedParserApi from '../../../api/feedParserApi';
import { toDateAndTimeFormat } from '../../../helpers/dateHelpers';
import ParseHistoryDetails from '../../../models/ParseHistoryDetails';
import ParseHistoryListing from '../../../models/ParseHistoryListing';
import { ButtonProps } from '../../common/button/Button';
import ButtonGroup from '../../common/button/ButtonGroup';
import ComponentLoading from '../../loading/ComponentLoading';

interface Props {
    parseHistoryItem: ParseHistoryListing;
}

const FeedParserHistoryItem: React.FC<Props> = ({ parseHistoryItem: summary }: Props): JSX.Element => {
    const [details, setDetails] = useState<ParseHistoryDetails | null>(null);
    const [forceResyncEnabled, setForceResyncEnabled] = useState<boolean>(false);
    const [isLoading, setIsLoading] = useState<boolean>(false);

    async function handleForceResyncClicked(): Promise<void> {
        if (details === null) return;
        setForceResyncEnabled(false);
        setIsLoading(true);
        await feedParserApi.forceSyncParseHistory(details.id);
        setIsLoading(false);
        setForceResyncEnabled(true);
    }

    const buttons = [
        {
            label: 'Force Re-sync',
            onClick: handleForceResyncClicked,
            enabled: forceResyncEnabled && (details?.episodeTitles?.length ?? 1 > 0),
        },
    ] as ButtonProps[];

    async function handlePanelExpansion(event: React.ChangeEvent<{}>, expanded: boolean): Promise<void> {
        if (!expanded) return;
        if (details !== null) return; // this is a re-expansion

        setDetails(await feedParserApi.getParseHistoryDetails(summary.id));
        setForceResyncEnabled(true);
    }

    const expansionPanelBody = (
        <ExpansionPanel
            data-testid={`feed-parser-history-item-${summary.id}`}
            TransitionProps={{ unmountOnExit: true }}
            onChange={handlePanelExpansion}
        >
            <ExpansionPanelSummary expandIcon={<span className="material-icons">expand_more</span>}>
                <div className="parse-history-summary">
                    <div className="date" title="Last synced on">
                        {toDateAndTimeFormat(moment(summary.lastSyncedOn))}
                    </div>
                    <div className="episode-count" title="Episode count">
                        {summary.episodeCount}
                    </div>
                    <div className="status">{summary.success ? 'Success' : 'Failure'}</div>
                </div>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails>
                <div className="parse-history-details">
                    {details === null ? (
                        <ComponentLoading />
                    ) : (
                        <ul>
                            {details?.episodeTitles
                                ?.sort((a, b) => (a > b ? 1 : -1))
                                .map((title) => (
                                    <li data-testid="episode-title" key={title}>
                                        {title}
                                    </li>
                                ))}
                        </ul>
                    )}
                </div>
            </ExpansionPanelDetails>
            <Divider />
            <ExpansionPanelActions>
                <ButtonGroup buttons={buttons} />
            </ExpansionPanelActions>
        </ExpansionPanel>
    );

    return <>{isLoading ? <ComponentLoading /> : expansionPanelBody}</>;
};

export default FeedParserHistoryItem;
