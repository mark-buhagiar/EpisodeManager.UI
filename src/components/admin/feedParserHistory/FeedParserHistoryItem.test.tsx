import {
    screen,
    act,
    cleanup,
    fireEvent,
    render,
    findByTestId,
    getByText,
    getAllByTestId,
    wait,
} from '@testing-library/react';
import moment from 'moment';
import React from 'react';
import { getParseHistoryDetails } from '../../../api/feedParserApi';
import FeedParserHistoryItem from './FeedParserHistoryItem';
import { toDateAndTimeFormat } from '../../../helpers/dateHelpers';

jest.mock('../../../api/feedParserApi', () => ({ getParseHistoryDetails: jest.fn() }));

const parseHistoryItem = {
    id: 28,
    retrievedOn: moment('2020-04-13T15:28:27.751938').toDate(),
    lastSyncedOn: moment('2020-04-13T15:28:27.751938').toDate(),
    episodeCount: 100,
    syncCount: 1,
    success: true,
};

const parseHistoryDetails = {
    id: 28,
    episodeTitles: [
        'Good Witch 6x01 The Anniversary 1080p',
        'MasterChef (AU) 12x12 1080p',
        'MasterChef (AU) 12x13 1080p',
        'MasterChef (AU) 12x14 Immunity Challenge: Time Auction 1080p',
        'MasterChef (AU) 12x15 Elimination Challenge: Texture 1080p',
        'Bar Rescue 7x10 Raging Turkey 720p',
        'American Idol 18x14 On with the Show: Homeward Bound',
        'The Rookie 2x19 The Q Word 1080p',
        'American Idol 18x14 On with the Show: Homeward Bound 720p',
        'Bar Rescue 7x10 Raging Turkey 1080p',
        'Good Girls 3x11 Synergy 1080p',
    ],
};

afterEach(async () => {
    await cleanup();
    jest.clearAllMocks();
});

describe('When the component is rendered', () => {
    it('should display a summary as the accordion header', async (): Promise<void> => {
        const { getByText, asFragment, getByTitle } = render(
            <FeedParserHistoryItem parseHistoryItem={parseHistoryItem} />,
        );
        expect(getByText(toDateAndTimeFormat(moment(parseHistoryItem.lastSyncedOn)))).toBeTruthy();
        expect(getByText(parseHistoryItem.success ? 'Success' : 'Failure')).toBeTruthy();
        expect(getByTitle('Episode count').textContent).toEqual(String(parseHistoryItem.episodeCount));
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('When the accordion header is clicked', () => {
    it('should display a loading indicator while additional details are being retrieved', async (): Promise<void> => {
        const { asFragment, getByRole } = render(<FeedParserHistoryItem parseHistoryItem={parseHistoryItem} />);
        (getParseHistoryDetails as jest.Mock).mockResolvedValue({});
        fireEvent.click(getByRole('button'));

        expect(asFragment()).toMatchSnapshot();
        expect(getByRole('progressbar')).toBeTruthy();
        await wait();
    });

    it('should display the retrieved details', async (): Promise<void> => {
        const { asFragment, getByRole, queryByRole, getAllByTestId } = render(
            <FeedParserHistoryItem parseHistoryItem={parseHistoryItem} />,
        );
        (getParseHistoryDetails as jest.Mock).mockResolvedValue({ ...parseHistoryDetails });

        fireEvent.click(getByRole('button'));
        await wait();

        const sortedEpisodeTitles = [...parseHistoryDetails.episodeTitles];
        sortedEpisodeTitles.sort((a, b) => (a > b ? 1 : -1));
        const episodesOnPage = getAllByTestId('episode-title');
        sortedEpisodeTitles.forEach((title, index) => expect(episodesOnPage[index].textContent).toEqual(title));
        expect(queryByRole('progressbar')).toBeNull();
        expect(asFragment()).toMatchSnapshot();
    });
});
