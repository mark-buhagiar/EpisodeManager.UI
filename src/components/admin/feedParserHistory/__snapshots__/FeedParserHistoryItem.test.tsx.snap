// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the accordion header is clicked should display a loading indicator while additional details are being retrieved 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiExpansionPanel-root Mui-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="feed-parser-history-item-28"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiExpansionPanelSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content Mui-expanded"
      >
        <div
          class="parse-history-summary"
        >
          <div
            class="date"
            title="Last synced on"
          >
            13 April 2020 15:28
          </div>
          <div
            class="episode-count"
            title="Episode count"
          >
            100
          </div>
          <div
            class="status"
          >
            Success
          </div>
        </div>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="material-icons"
          >
            expand_more
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container"
      style="min-height: 0px; height: 0px; transition-duration: 0ms;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiExpansionPanelDetails-root"
            >
              <div
                class="parse-history-details"
              >
                <div
                  class="component-loading-container"
                >
                  <span
                    class="component-loading"
                  >
                    <div
                      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiExpansionPanelActions-root MuiExpansionPanelActions-spacing"
            >
              <div
                class="button-group "
              >
                <button
                  class="button no-select disabled"
                  data-testid="button-Force Re-sync"
                  role="disabled-button"
                  title="Disabled"
                  type="button"
                >
                  <div
                    class="disabled-overlay"
                  />
                  Force Re-sync
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`When the accordion header is clicked should display the retrieved details and the force sync button 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiExpansionPanel-root Mui-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="feed-parser-history-item-28"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiExpansionPanelSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content Mui-expanded"
      >
        <div
          class="parse-history-summary"
        >
          <div
            class="date"
            title="Last synced on"
          >
            13 April 2020 15:28
          </div>
          <div
            class="episode-count"
            title="Episode count"
          >
            100
          </div>
          <div
            class="status"
          >
            Success
          </div>
        </div>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="material-icons"
          >
            expand_more
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px; height: auto; transition-duration: 0ms;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiExpansionPanelDetails-root"
            >
              <div
                class="parse-history-details"
              >
                <ul>
                  <li
                    data-testid="episode-title"
                  >
                    American Idol 18x14 On with the Show: Homeward Bound
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    American Idol 18x14 On with the Show: Homeward Bound 720p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    Bar Rescue 7x10 Raging Turkey 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    Bar Rescue 7x10 Raging Turkey 720p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    Good Girls 3x11 Synergy 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    Good Witch 6x01 The Anniversary 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    MasterChef (AU) 12x12 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    MasterChef (AU) 12x13 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    MasterChef (AU) 12x14 Immunity Challenge: Time Auction 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    MasterChef (AU) 12x15 Elimination Challenge: Texture 1080p
                  </li>
                  <li
                    data-testid="episode-title"
                  >
                    The Rookie 2x19 The Q Word 1080p
                  </li>
                </ul>
              </div>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiExpansionPanelActions-root MuiExpansionPanelActions-spacing"
            >
              <div
                class="button-group "
              >
                <button
                  class="button no-select "
                  data-testid="button-Force Re-sync"
                  role="button"
                  title=""
                  type="button"
                >
                  Force Re-sync
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`When the component is rendered should display a summary as the accordion header 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="feed-parser-history-item-28"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiExpansionPanelSummary-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content"
      >
        <div
          class="parse-history-summary"
        >
          <div
            class="date"
            title="Last synced on"
          >
            13 April 2020 15:28
          </div>
          <div
            class="episode-count"
            title="Episode count"
          >
            100
          </div>
          <div
            class="status"
          >
            Success
          </div>
        </div>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="material-icons"
          >
            expand_more
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
